src/services/rsvp/rsvp-router.ts

```diff
@@ -0,0 +1,154 @@
+import express, { Request, Response, Router } from "express";
+import Constants from "../../constants.js";
+import { strongJwtVerification } from "../../middleware/verify-jwt.js";
+import { JwtPayload } from "../auth/auth-models.js";
+import { hasElevatedPerms } from "../auth/auth-lib.js";
+import { DecisionStatus, DecisionResponse, DecisionInfo } from "../../database/decision-db.js";
+import Models from "../../database/models.js";
+
+const rsvpRouter: Router = Router();
+rsvpRouter.use(express.urlencoded({ extended: false }));
+
+rsvpRouter.get("/test/", (_: Request, res: Response) => {
+    res.end("RSVP endpoint is working!");
+});
+
+/**
+ * @api {get} /rsvp/:USERID/ GET /rsvp/:USERID/
+ * @apiGroup rsvp
+ * @apiDescription Check RSVP decision for a given userId, provided that the current user has elevated perms
+ *
+ * @apiSuccess (200: Success) { string, boolean } usedId and whether they are/aren't attending
+ * @apiSuccessExample Example Success Response:
+ * 	HTTP/1.1 200 OK
+ *	{
+ *		"userId": "github0000001",
+ *      "isAttending": true
+ * 	}
+ *
+ * @apiUse strongVerifyErrors
+ */
+rsvpRouter.get("/:USERID", strongJwtVerification, async (req: Request, res: Response) => {
+    const userId: string | undefined = req.params.USERID;
+
+    //Returns error if userid parameter is empty
+    if (!userId) {
+        return res.status(Constants.BAD_REQUEST).send({ error: "InvalidParams" });
+    }
+
+    const payload: JwtPayload = res.locals.payload as JwtPayload;
+    //Returns error if caller doesn't have elevated perms
+    if (!hasElevatedPerms(payload)) {
+        return res.redirect("/");
+    }
```

Comment:
What if this is called on your own userId? i.e. `payload.id == userId`?

NVM, this still works